### PR TITLE
Fix more C++ lints in cpp

### PR DESCRIPTION
**Pull request by bernardnormier**
These lints were fixed "by hand".

### CODE REVIEW

File: cpp/include/Ice/Proxy.h [hunk 1]
```diff
@@ -259,8 +259,8 @@ namespace Ice
 
         // The copy constructor and assignment operators are to keep GCC happy.
         Proxy(const Proxy&) noexcept = default;
-        Proxy& operator=(const Proxy&) noexcept { return *this; }
-        Proxy& operator=(Proxy&&) noexcept { return *this; }
+        Proxy& operator=(const Proxy&) noexcept { return *this; } // NOLINT
```
Comment (externl) [inline cpp/include/Ice/Proxy.h:262]: Can we include the lint name like we're doing in other places.

File: cpp/src/Glacier2CryptPermissionsVerifier/CryptPermissionsVerifierI.cpp [hunk 2]
```diff
@@ -107,9 +107,11 @@ namespace
     {
 #if defined(__APPLE__) || defined(_WIN32)
 
+        // NOLINTBEGIN:cert-err58-cpp
```
Comment (externl) [inline cpp/src/Glacier2CryptPermissionsVerifier/CryptPermissionsVerifierI.cpp:110]: Can we just get rid of the constants? They're only used in this function. and we make a copy all the time.

File: cpp/src/Ice/Exception.cpp [hunk 3]
```diff
@@ -15,7 +15,7 @@
 // For UINTPTR_MAX on Ubuntu Precise
 //
 #ifndef __STDC_LIMIT_MACROS
-#    define __STDC_LIMIT_MACROS
+#    define __STDC_LIMIT_MACROS // NOLINT
```
Comment (externl) [inline cpp/src/Ice/Exception.cpp:18]: Ditto. What doesn't the linter like here?

**Review by externl — APPROVED** [review]
Looks good!

File: cpp/include/Ice/Proxy.h (hunk 1 above)
Comment (bernardnormier) [reply cpp/include/Ice/Proxy.h:262]: These are dummy functions to keep GCC happy. Here, clang-tidy is complaining about not testing for self-assignment, which is irrelevant.

File: cpp/src/Ice/Exception.cpp (hunk 3 above)
Comment (bernardnormier) [reply cpp/src/Ice/Exception.cpp:18]: It doesn't like that we define a __XXX symbol.

File: cpp/src/Glacier2CryptPermissionsVerifier/CryptPermissionsVerifierI.cpp (hunk 2 above)
Comment (bernardnormier) [reply cpp/src/Glacier2CryptPermissionsVerifier/CryptPermissionsVerifierI.cpp]: Fixed. In simple cases like these, we should use `const char* const`

File: cpp/src/DataStorm/TraceUtil.h [hunk 4]
```diff
@@ -11,7 +11,8 @@
 #include "SessionI.h"
 #include "TopicI.h"
 
-namespace std
+// TODO: explain why we need to use namespace std here.
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply cpp/src/DataStorm/TraceUtil.h:14]: Probably, but it's not immediately clear to me.

**Pull request merged**